v2/slogger/rolling_file_appender/rolling_file_appender.go

```diff
@@ -127,7 +160,10 @@ func (self *RollingFileAppender) Append(log *slogger.Log) error {
 		return err
 	}
 
-	if self.MaxFileSize > 0 && self.curFileSize > self.MaxFileSize {
+	if (self.maxFileSize > 0 && self.curFileSize > self.maxFileSize) ||
+		(self.maxDuration > 0 &&
+			self.state != nil &&
```

Comment:
I agree.  It was just added as a sanity check, particularly if the code should change to allow it to happen.